index.bs

```diff
@@ -583,36 +583,30 @@ interface ML {
 };
 </script>
 
-The {{ML/createContext()}} method steps are:
-1. If [=this=]'s [=relevant global object=]'s [=associated Document=] is not [=allowed to use=] the [=webnn-feature|webnn=] feature, then throw a "{{SecurityError!!exception}}" {{DOMException}} and abort these steps.
-1. Let |promise| be [=a new promise=].
-1. Let |context| be a new {{MLContext}} object.
-1. Switch on the method's first argument:
-    <dl class=switch>
-    <dt>{{MLContextOptions}}
-    <dd>Set |context|.{{[[contextType]]}} to [=default-context|default=].
-    <dd>Set |context|.{{[[deviceType]]}} to the value of {{MLContextOptions}}'s {{deviceType}}.
-    <dd>Set |context|.{{[[powerPreference]]}} to the value of {{MLContextOptions}}'s {{powerPreference}}.
-
-    <dt>{{GPUDevice}}
-    <dd>Set |context|.{{[[contextType]]}} to [=webgpu-context|webgpu=].
-    <dd>Set |context|.{{[[deviceType]]}} to "[=device-type-gpu|gpu=]".
-    <dd>Set |context|.{{[[powerPreference]]}} to "[=power-preference-default|default=]".
-    </dl>
-1. Issue the following steps to a separate timeline:
-    1. If the User Agent can support the |context|.{{[[contextType]]}}, |context|.{{[[deviceType]]}} and |context|.{{[[powerPreference]]}}, then:
-        1. Set |context|.{{MLContext/[[implementation]]}} to an implementation supporting |context|.{{[[contextType]]}}, |context|.{{[[deviceType]]}} and |context|.{{[[powerPreference]]}}.
-        1. [=Resolve=] |promise| with |context|.
-    1. Else:
-        1. [=Resolve=] |promise| with a new {{NotSupportedError}}.
-1. Return |promise|.
-
 ### Permissions Policy Integration ### {#permissions-policy-integration}
 
 This specification defines a <a>policy-controlled feature</a> identified by the
 string "<code><dfn data-lt="webnn-feature">webnn</dfn></code>".
 Its <a>default allowlist</a> is <code>'self'</code>.
 
+### The {{ML/createContext()}} method ### {#api-ml-createcontext}
+The {{ML/createContext()}} method steps are:
+1. If [=this=]'s [=relevant global object=]'s [=associated Document=] is not [=allowed to use=] the [=webnn-feature|webnn=] feature, return [=a new promise=] [=rejected=] with a "{{SecurityError}}" and abort these steps.
+1. Let |promise| be [=a new promise=].
+1. Return |promise| and run the following steps [=in parallel=].
+1. Let |context| be a new {{MLContext}} object.
+1. Let |options| be the first argument.
+1. If |options| is a {{GPUDevice}},
+    1. Set |context|.{{[[contextType]]}} to "[=webgpu-context|webgpu=]".
+    1. Set |context|.{{[[deviceType]]}} to "[=device-type-gpu|gpu=]".
+    1. Set |context|.{{[[powerPreference]]}} to "[=power-preference-default|default=]".
+1. Otherwise,
+    1. Set |context|.{{[[contextType]]}} to "[=default-context|default=].
+    1. Set |context|.{{[[deviceType]]}} to the value of |options|'s {{deviceType}} if defined, or to "[=device-type-cpu|cpu=]".
```

Comment:
[Dictionary](https://webidl.spec.whatwg.org/#idl-dictionary) options is an ordered map, so we should use "map[key] [exists](https://infra.spec.whatwg.org/#map-exists)" syntax for this and other similar checks.

You should also check that we use "exists" consistently in the spec. I spotted a few occurrences of "exist" that do not link to the dfn in the MLCommandEncoder interface. Spin a new PR for those.

Reply:
I keep seeing the following pattern being used with dictionaries:

    1. If options["deviceType"] exists, then
      1. Set context[[deviceType]] to options["deviceType"]. 
    1. Otherwise, set context[[deviceType]] to "cpu".

See e.g. https://www.w3.org/TR/generic-sensor/#abstract-operations

Reply:
Equivalent to the current text, 

> 1. Set |context|.{{[[powerPreference]]}} to the value of |options|'s {{powerPreference}} if it [=map/exists=], or to "[=power-preference-default|default=]".

But I can follow the other style at the expense of verbosity, no problem.

Reply:
BTW nowadays we are not supposed to use `options["deviceType"]`, we should say `options`'  `deviceType` (link), or 
"if `options` includes `deviceType`". Same applies to |context| etc. 

Need to check more specifications.

Reply:
BTW nowadays we are not supposed to use `options["deviceType"]`, we should say `options`'  `deviceType` (link), or 
"if `options` includes `deviceType`". Same applies to |context| etc. 

Need to check more specifications.

Reply:
Thanks @zolkis for the research, in my experience conventions in the infra spec are safe to use. Given that, I think you want to change this line to something like:

```
1. If options["deviceType"] exists, then
  1. Set context[[deviceType]] to options["deviceType"]. 
1. Otherwise, set context[[deviceType]] to "cpu".
```

And similarly on the following line for powerPreference.

Reply:
It's more fluffy and repeating stuff than saying, 

> If |options|'s {{powerPreference}} [=map/exists=], set |context|.{{[[powerPreference]]}} to its value, otherwise to "[=power-preference-default|default=]".

but let it be as you wish.

Reply:
The one line version is better in this context. To align with the convention, you should replace "its value" with "options["deviceType"]":

`1. If options["deviceType"] exists, then set context[[deviceType]] to options["deviceType"]. Otherwise, set context[[deviceType]] to "cpu".`

Reply:
I just checked a couple of other specs, and the convention they use is 2 steps, one for "if" and one for "otherwise", just as the first suggestion in my [previous](https://github.com/webmachinelearning/webnn/pull/310#discussion_r1045580031) comment.

Reply:
Not wishing to over-stretch the issue, but avoiding unnecessary repetitions would make the text more clear IMHO.
What about using

```1. If options["deviceType"] exists, then set context[[deviceType]] to its value. Otherwise, set context[[deviceType]] to "cpu".```

Reply:
I do also think that reads more like English, but if we want to use the infra spec conventions consistently we should replace `its value` with `options["deviceType"]`.

Note that map get is with quote marks and map set without:

https://infra.spec.whatwg.org/#map-get 
https://infra.spec.whatwg.org/#map-set

It is not over-stretching. It is important to agree on these conventions so we can then apply the conventions in follow ups consistently. My recommendation would be to align with the infra spec conventions whenever available.

Reply:
No problem, it's already written that way (assumed we will stick to the more verbose standard pattern).